test/system/src/test/java/io/pravega/test/system/MultiReaderWriterTxnWithFailoverTest.java

```diff
@@ -0,0 +1,444 @@
+            //create writers
+            log.info("Creating {} writers", NUM_WRITERS);
+            List<EventStreamWriter<Long>> writerList = new ArrayList<>();
+            log.info("Writers writing in the scope {}", scope);
+            for (int i = 0; i < NUM_WRITERS; i++) {
+                log.info("Starting writer{}", i);
+                final EventStreamWriter<Long> writer = clientFactory.createEventWriter(STREAM_NAME,
+                        new JavaSerializer<Long>(),
+                        EventWriterConfig.builder().retryAttempts(10).build());
+                writerList.add(writer);
+            }
+
+            //create a reader group
+            log.info("Creating Reader group : {}", readerGroupName);
+            log.info("Scope passed to readergroup manager {}", scope);
+
+            readerGroupManager.createReaderGroup(readerGroupName, ReaderGroupConfig.builder().startingTime(0).build(),
+                    Collections.singleton(STREAM_NAME));
+            log.info("Reader group name {} ", readerGroupManager.getReaderGroup(readerGroupName).getGroupName());
+            log.info("Reader group scope {}", readerGroupManager.getReaderGroup(readerGroupName).getScope());
+            log.info("Online readers {}", readerGroupManager.getReaderGroup(readerGroupName).getOnlineReaders());
+
+            //create readers
+            log.info("Creating {} readers", NUM_READERS);
+            List<EventStreamReader<Long>> readerList = new ArrayList<>();
+            String readerName = "reader" + new Random().nextInt(Integer.MAX_VALUE);
+            log.info("Scope that is seen by readers {}", scope);
+            //start reading events
+            for (int i = 0; i < NUM_READERS; i++) {
+                log.info("Starting reader{}", i);
+                log.info("Creating reader with id {}", readerName + i);
+                final EventStreamReader<Long> reader = clientFactory.createReader(readerName + i,
+                        readerGroupName,
+                        new JavaSerializer<Long>(),
+                        ReaderConfig.builder().build());
+                readerList.add(reader);
+            }
+
+            // start writing asynchronously
+            List<CompletableFuture<Void>> writerFutureList = new ArrayList<>();
+            writerList.forEach(writer -> {
+                CompletableFuture<Void> writerFuture = startWritingIntoTxn(eventData, writer, stopWriteFlag, eventWriteCount);
+                writerFutureList.add(writerFuture);
+            });
+
+            //start reading asynchronously
+            List<CompletableFuture<Void>> readerFutureList = new ArrayList<>();
+            readerList.forEach(reader -> {
+                CompletableFuture<Void> readerFuture = startReading(eventsReadFromPravega, eventWriteCount, eventReadCount, stopReadFlag, reader);
+                readerFutureList.add(readerFuture);
+            });
+
+            //perform the scaling operations
```

Comment:
It is a bit odd to say `perform scaling...` and call the method `perform failover...`, it sounds like the comment and the method call don't match.